### DISCUSSION TITLE
beta.96 - Design, Drop, and Hotkeys! Oh my!

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#150] [Announcements]
https://user-images.githubusercontent.com/36073/115079813-fc5f2200-9ebe-11eb-8e7c-74c8a1d2aee3.mp4

Can't wait to see what you build! Happy Scripting this weekend! 😇

**Comment by peterjohnhunt** [comment]
This is awesome!

Re: the "hotkey" option, I was thinking last night that'd be cool to show a todo item and be able to: Mark as complete, edit title, re-assign, etc based on a shortcut when viewing. Excited to try it out!

**Reply by johnlindquist** [reply]
Nice, can't wait to see what you come up with 👀

**Comment by jsjoeio** [comment]
This is awesome! I already have a couple ideas just from the demo. 

Re: drop - will it take something from my clipboard or is that a v2 feature?

**Reply by johnlindquist** [reply]
Like if you copy an image to your clipboard? No, but I'll see what I can do soon. 👍

**Reply by johnlindquist** [reply]
Actually, if you can explain your scenario a bit more, that would be helpful. Would you expect an image buffer returned?

**Reply by jsjoeio** [reply]
Hmm...I'm not fully sure. 

I think I'm confusing this drop with this one I see you use [here](https://github.com/johnlindquist/kit/discussions/100)

Basically, I like this `drop` you demoed with beta.96 (I like that it waits, I like the dotted lines, etc) and was wondering if this `drop` could support an image from clipboard for a similar use case to your "Rotate Dropped Image" script.

**Reply by johnlindquist** [reply]
Oh, it's the same `drop` as the one in #80, but wrapped in a `drop` helper and enhanced to support all the drop types and ignore blur.

The "rotate" script works on the assumption that you get a file _path_ from what you drop. Then the script reads the file path to display it.

Using the clipboard (as in right-click, "copy image to clipboard" in Chrome) gives image data (not a path), so it becomes a little more involved to handle the image data. I would probably expose that through a `getClipboardImage` or something api, but I'd have to think through the scenarios to better understand the end goals.

**Reply by jsjoeio** [reply]
Ah, I understand now!

Okay, that makes more sense. No immediate needs I don't want to steal your attention for that, but if/when it comes up, I'll be sure to share and get your thoughts. Thanks for explaining that!
